### ISSUE TITLE
Microsoft.VisualStudio.Setup.Configuration.* - License

### ISSUE THREAD

**Issue by 3F** [Microsoft/vs-setup-samples#11]
Did I understand correctly that https://go.microsoft.com/fwlink/?linkid=831109 (MICROSOFT SOFTWARE LICENSE TERMS - MICROSOFT VISUAL STUDIO SETUP SOFTWARE DEVELOPMENT KIT) for packages:

* Microsoft.VisualStudio.Setup.Configuration.Native
* Microsoft.VisualStudio.Setup.Configuration.Interop

Are fully incompatible with [MIT](https://opensource.org/licenses/MIT) ? *well, GPL, and other from OSI*

You may not ... 
```
reverse engineer, decompile or disassemble the software, .... 
share, publish, rent or lease the software, or provide the software as a stand-alone hosted as solution for others to use.
```

... you must:
```
- add significant primary functionality to it in your applications;
- require distributors and external end users to agree to terms that protect the Distributable Code at least as much as this agreement
```

```
DISTRIBUTABLE CODE ....the term “distribution” also means deployment of your applications for third parties to access 
```

your [vswhere](https://github.com/Microsoft/vswhere) provides a more flexible rights, so have you plan to change license for this simple packages ? Or how to legally use it without terms to agree for end users, and so on.

I'm not a guru with licensing, just confused how to apply MIT license if your .lib will be included inside X program o_O

**Comment by heaths** [comment]
_vswhere_ is redistributable while the DLLs in question in their current form aren't **intended** to be. Originally I had planned that they would be, but since anyone redistributing them may not frequently enough update them older versions may no longer work. You really don't even need the _.lib_ if you use `uuidof(ISetupConfiguration)` (for example). The license is mainly to allow static linking and interop assembly embedding (so you don't actually have to ship the assembly DLL).

Though if you have concerns about licensing with your application, I recommend you discuss it with independent counsel.

**Comment by 3F** [comment]
Thanks for the clarification.

> The license is mainly to allow static linking and interop assembly embedding 

I'm still confused about static linkage because of some items from this license -_-

> Originally I had planned that they would be, but since anyone redistributing them may not frequently enough update them older versions may no longer work.

I see, thanks.

> I recommend you discuss it with independent counsel.

Well, in recent time, MS wants to be more open or more accessible in open source, but periodically we have (I think community at all) some questions about legal usage their components for some cases. Especially like from VSSDK etc. And the license like [this](https://go.microsoft.com/fwlink/?linkid=831109) is really a little harder to understand what to do at least for developers. Does Microsoft provide some special online resource / forum to get additional clarification about legal usage of some this things together with OSI and so on ?

**Comment by heaths** [comment]
You are allowed to link both the native lib and managed assembly, and since the MIT licence is not viral like the GPL or LGPL I don't see a problem, but I also am not a lawyer. But I did work with our lawyers on the licensing of this software.

I don't know of any resources likes you're asking about and I can ask internally but an answer may take some time. I recommended independent counsel because in legal matters - in general - it's a good idea.
